website/src/components/AddressExplorerLink.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useEffect, useState } from "react";
+
+type ChainID = 1 | 5 | 42170 | 421613 | 42161;
+
+const chainIDToExplorerUrlRool: {
+  [chainId in ChainID]: string;
+} = {
+  1: "https://etherscan.io/address",
+  5: "https://goerli.etherscan.io/address",
+  42161: "https://arbiscan.io/address",
+  42170: "https://nova.arbiscan.io/address",
+  421613: "https://goerli.arbiscan.io/address",
+};
+
+export const AddressExplorerLink = (props: {
+  address: string;
+  chainID: ChainID;
+}) => {
+  const { address, chainID } = props;
+  const rootUrl = chainIDToExplorerUrlRool[chainID];
+  // if(!rootUrl) throw new Error(`Error: no root url set for chain id ${chainID} `)
```

Comment:
remove comment?

Reply:
yeah i was hoping typescript would catch invalid inputs statically (i.e., unsupported chain id), but it doesn't; with this error, the page crashes, but the build doesn't fail. I think the page crashing is obvious enough / better than nothing, so uncommenting